### PR TITLE
Admin custom email

### PR DESCRIPTION
**Pull request by dreamfall**

### CODE REVIEW

File: app/assets/javascripts/frontend/textarea-autoResize.js [hunk 1]
```diff
@@ -26,6 +26,10 @@ $(function() {
     var char_count = content.toString().length;
     var new_line_count = content.split("\n").length;
     var text_width = char_count * 8;
+    if (!box_width) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/textarea-autoResize.js]: Identifier 'box_width' is not in camel case.

File: app/assets/javascripts/frontend/textarea-autoResize.js [hunk 2]
```diff
@@ -26,6 +26,10 @@ $(function() {
...
+      box_width = textInput.innerWidth();
```
Comment (matthewford) [inline app/assets/javascripts/frontend/textarea-autoResize.js]: Identifier 'box_width' is not in camel case.

File: app/assets/javascripts/frontend/textarea-autoResize.js [hunk 3]
```diff
@@ -26,6 +26,10 @@ $(function() {
...
+      text_width = char_count * 7;
```
Comment (matthewford) [inline app/assets/javascripts/frontend/textarea-autoResize.js]: Identifier 'text_width' is not in camel case.<br>Identifier 'char_count' is not in camel case.

File: app/assets/stylesheets/admin/page-custom-emails.scss [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    padding: 4em 0;
+    padding-bottom: 7em;
+
+    textarea {
+      padding: 0 12px;
+      border: none;
+      background: transparent;
+      resize: none;
+      -webkit-box-shadow: none;
+              box-shadow: none;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/page-custom-emails.scss]: Line should be indented 6 spaces, but was indented 14 spaces

File: app/assets/stylesheets/admin/page-custom-emails.scss [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  .email-container {
+    position: relative;
+    padding: 4em 0;
+    padding-bottom: 7em;
+
+    textarea {
+      padding: 0 12px;
+      border: none;
+      background: transparent;
+      resize: none;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/page-custom-emails.scss]: Properties should be ordered padding, border, background, -webkit-box-shadow, box-shadow, resize

File: app/assets/stylesheets/admin/page-custom-emails.scss [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      -webkit-box-shadow: none;
+              box-shadow: none;
+    }
+  }
+
+  .email-header,
+  .email-footer {
+    position: absolute;
+    padding: 10px 12px;
+    font-size: 14px;
```
Comment (matthewford) [inline app/assets/stylesheets/admin/page-custom-emails.scss]: Properties should be ordered position, padding, color, font-size, line-height

File: app/models/user.rb [hunk 7]
```diff
@@ -56,6 +56,8 @@ def after_initialize
       where.not(id: user.id)
     }
     scope :by_email, -> { order(:email) }
+    scope :qae_opt_in_group, -> { where(subscribed_to_emails: true) }
```
Comment (matthewford) [inline app/models/user.rb]: Inconsistent indentation detected.

File: app/models/user.rb [hunk 8]
```diff
@@ -56,6 +56,8 @@ def after_initialize
...
+    scope :bit_opt_in, -> { where(agree_being_contacted_by_department_of_business: true) }
```
Comment (matthewford) [inline app/models/user.rb]: Inconsistent indentation detected.

File: spec/forms/custom_email_form_spec.rb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  let!(:admin) { create :admin }
+
+  it "sends email to myself" do
+    form = CustomEmailForm.new(user: admin, scope: "myself", message: "message", subject: "subject")
+    expect(form.emails).to eq([admin.email])
+  end
+
+  it "sends email to users with qae_opt_in_group" do
+    user_1 = create(:user, subscribed_to_emails: true)
+    user_2 = create(:user, subscribed_to_emails: false)
```
Comment (matthewford) [inline spec/forms/custom_email_form_spec.rb]: Useless assignment to variable - `user_2`. Did you mean `user_1`?

File: spec/forms/custom_email_form_spec.rb [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    user_3 = create(:user, subscribed_to_emails: true)
+
+    form = CustomEmailForm.new(user: admin, scope: "qae_opt_in_group", message: "message", subject: "subject")
+    expect(form.emails).to eq([user_1.email, user_3.email])
+  end
+
+  it "sends email to users with bis_opt_in" do
+    user_1 = create(:user, agree_being_contacted_by_department_of_business: true)
+    user_2 = create(:user, agree_being_contacted_by_department_of_business: false)
```
Comment (matthewford) [inline spec/forms/custom_email_form_spec.rb]: Useless assignment to variable - `user_2`. Did you mean `user_1`?

File: spec/forms/custom_email_form_spec.rb [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    form = CustomEmailForm.new(user: admin, scope: "myself", message: "message", subject: "subject")
+    expect(form.emails).to eq([admin.email])
+  end
+
+  it "sends email to users with qae_opt_in_group" do
+    user_1 = create(:user, subscribed_to_emails: true)
+    user_2 = create(:user, subscribed_to_emails: false)
+    user_3 = create(:user, subscribed_to_emails: true)
+
+    form = CustomEmailForm.new(user: admin, scope: "qae_opt_in_group", message: "message", subject: "subject")
```
Comment (matthewford) [inline spec/forms/custom_email_form_spec.rb]: Line is too long. [110/100]

File: spec/forms/custom_email_form_spec.rb [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    expect(form.emails).to eq([user_1.email, user_3.email])
+  end
+
+  it "sends email to users with bis_opt_in" do
+    user_1 = create(:user, agree_being_contacted_by_department_of_business: true)
+    user_2 = create(:user, agree_being_contacted_by_department_of_business: false)
+    user_3 = create(:user, agree_being_contacted_by_department_of_business: true)
+
+    form = CustomEmailForm.new(user: admin, scope: "bis_opt_in", message: "message", subject: "subject")
```
Comment (matthewford) [inline spec/forms/custom_email_form_spec.rb]: Line is too long. [104/100]

File: spec/forms/custom_email_form_spec.rb [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    expect(form.emails).to eq([user_1.email, user_3.email])
+  end
+
+  it "sends email to all assessors" do
+    user_1 = create(:assessor)
+    user_2 = create(:assessor)
+    user_3 = create(:assessor)
+
+    form = CustomEmailForm.new(user: admin, scope: "assessors", message: "message", subject: "subject")
```
Comment (matthewford) [inline spec/forms/custom_email_form_spec.rb]: Line is too long. [103/100]

File: spec/forms/custom_email_form_spec.rb [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    expect(form.emails).to eq([user_1.email, user_2.email, user_3.email])
+  end
+
+  it "sends email to all users" do
+    user_1 = create(:user)
+    user_2 = create(:user)
+    user_3 = create(:user)
+
+    form = CustomEmailForm.new(user: admin, scope: "all_users", message: "message", subject: "subject")
```
Comment (matthewford) [inline spec/forms/custom_email_form_spec.rb]: Line is too long. [103/100]

**Comment by RuslanHamidullin** [comment]
Good to merge :+1:

**Pull request merged**